### DISCUSSION TITLE
Next.js - cannot import or require() from ./src/... into tailwind.config

### DISCUSSION THREAD

**Discussion by Johnrobmiller** [tailwindlabs/tailwindcss#8385] [Help]
It is best practice for a design system to only have one source of truth for a color scheme. The colors might be stored in a ./constants file, or maybe a dedicated file for colors, or something like that.

However, it is not possible in Next.js to require() in these colors into tailwind.config! This makes it impossible to have a single source of truth for colors in the app, as it is not realistic for devs to expect to use tailwind for absolutely everything.

This is a major issue, as without a single source of truth for colors, the codebase I'm working on increases in complexity and becomes harder to maintain or change.

After some googling, I see that this has been a problem or years. Any possibility of fixing this sometime soon?

**Comment by adamwathan** [comment]
Hey! Please don't create issues without following the issue template.

I've tried to reproduce what you've outlined here by hand and it all seems to work to me:

https://github.com/adamwathan/next-color-import-test

Even if it didn't I'm afraid this wouldn't be a Tailwind bug, just a CJS/ESM interop issue.
